Gui900.cpp

```diff
@@ -1,1169 +1,1144 @@
+    initialize2dArray(button_text,pageCount,buttonCount);
+    initialize2dArray(button_textColor,pageCount,buttonCount);
+    initialize2dArray(button_backColor,pageCount,buttonCount);
+    initialize2dArray(button_border,pageCount,buttonCount);
+    initialize2dArray(button_enabled,pageCount,buttonCount);
+    initialize2dArray(button_visible,pageCount,buttonCount);
+    initialize2dArray(button_radius,pageCount,buttonCount);
+    initialize2dArray(button_XTO,pageCount,buttonCount);
+    initialize2dArray(button_YTO,pageCount,buttonCount);
+    initialize2dArray(button_pressed,pageCount,buttonCount);
+
+    
+    initializeArray(label_counts,pageCount);
+
+    initialize2dArray(label_Xpos,pageCount,labelCount);
+    initialize2dArray(label_Ypos,pageCount,labelCount);
+    initialize2dArray(label_text,pageCount,labelCount);
+    initialize2dArray(label_textColor,pageCount,labelCount);
+    initialize2dArray(label_textSizeX,pageCount,labelCount);
+    initialize2dArray(label_textSizeY,pageCount,labelCount);
+    initialize2dArray(label_enabled,pageCount,labelCount);
+    initialize2dArray(label_visible,pageCount,labelCount);
+
+    label_font=new GFXfont**[pageCount];
+    for(int i=0;i<labelCount;i++){
+        label_font[i]=new GFXfont*[labelCount];
+    }
+    
+
+    initializeArray(checkbox_counts,pageCount);
+
+    initialize2dArray(checkbox_Xpos,pageCount,checkBoxCount);
+    initialize2dArray(checkbox_Ypos,pageCount,checkBoxCount);
+    initialize2dArray(checkbox_text,pageCount,checkBoxCount);
+    initialize2dArray(checkbox_size,pageCount,checkBoxCount);
+    initialize2dArray(checkbox_checkColor,pageCount,checkBoxCount);
+    initialize2dArray(checkbox_textColor,pageCount,checkBoxCount);
+    initialize2dArray(checkbox_boxColor,pageCount,checkBoxCount);
+    initialize2dArray(checkbox_checked,pageCount,checkBoxCount);
+    initialize2dArray(checkbox_enabled,pageCount,checkBoxCount);
+    initialize2dArray(checkbox_visible,pageCount,checkBoxCount);
+    initialize2dArray(checkbox_lastState,pageCount,checkBoxCount);
+
+
+    initializeArray(slider_counts,pageCount);
+
+    initialize2dArray(slider_Xpos,pageCount,sliderCount);
+    initialize2dArray(slider_Ypos,pageCount,sliderCount);
+    initialize2dArray(slider_color_border,pageCount,sliderCount);
+    initialize2dArray(slider_color_left,pageCount,sliderCount);
+    initialize2dArray(slider_color_right,pageCount,sliderCount);
+    initialize2dArray(slider_color_thumb,pageCount,sliderCount);
+    initialize2dArray(slider_height,pageCount,sliderCount);
+    initialize2dArray(slider_width,pageCount,sliderCount);
+    initialize2dArray(slider_touch_area_height,pageCount,sliderCount);
+    initialize2dArray(slider_thumb_width,pageCount,sliderCount);
+    initialize2dArray(slider_value,pageCount,sliderCount);
+    initialize2dArray(slider_enabled,pageCount,sliderCount);
+    initialize2dArray(slider_visible,pageCount,sliderCount);
+    
+}
```

Comment:
This code is causing the micro-controller to crash.
The commands are not efficient. It contains about a thousand function calls.

Reply:
Here, the pointers were assigned in function scope. It is fixed.